osu.Game/Overlays/Wiki/WikiSidebar.cs

```diff
@@ -0,0 +1,116 @@
+        {
+            Direction = FillDirection.Vertical,
+            RelativeSizeAxes = Axes.X,
+            AutoSizeAxes = Axes.Y,
+            Children = new Drawable[]
+            {
+                new OsuSpriteText
+                {
+                    Text = "CONTENTS",
+                    Font = OsuFont.GetFont(size: 12, weight: FontWeight.Bold),
+                },
+                tableOfContents = new FillFlowContainer
+                {
+                    Direction = FillDirection.Vertical,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                }
+            },
+        };
+
+        public void AddToc(HeadingBlock headingBlock, MarkdownHeading heading)
+        {
+            switch (headingBlock.Level)
+            {
+                case 2:
+                case 3:
+                    string title = getTitle(headingBlock.Inline);
+                    tableOfContents.Add(new TocTitle(title, heading, headingBlock.Level == 3));
+                    break;
+            }
+        }
+
+        private string getTitle(ContainerInline containerInline)
+        {
+            foreach (var inline in containerInline)
+            {
+                switch (inline)
+                {
+                    case LiteralInline literalInline:
+                        return literalInline.Content.ToString();
+
+                    case LinkInline { IsImage: false } linkInline:
+                        return getTitle(linkInline);
+                }
+            }
+
+            return string.Empty;
+        }
+
+        private class TocTitle : OsuHoverContainer
+        {
+            [Resolved]
+            private OverlayScrollContainer scrollContainer { get; set; }
+
+            private readonly MarkdownHeading target;
+
+            private readonly OsuTextFlowContainer textFlow;
+
+            public TocTitle(string text, MarkdownHeading target, bool subtitle = false)
+            {
+                this.target = target;
+
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+                Child = textFlow = new OsuTextFlowContainer(t =>
+                {
+                    t.Font = OsuFont.GetFont(size: subtitle ? 12 : 15);
+                }).With(f =>
+                {
+                    f.AddText(text);
+                    f.RelativeSizeAxes = Axes.X;
+                    f.AutoSizeAxes = Axes.Y;
+                });
+                Margin = new MarginPadding { Top = subtitle ? 5 : 10 };
+                Padding = new MarginPadding { Left = subtitle ? 10 : 0 };
+            }
+
+            protected override IEnumerable<Drawable> EffectTargets => new Drawable[] { textFlow };
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider)
+            {
+                IdleColour = colourProvider.Light2;
+                HoverColour = colourProvider.Light1;
```

Comment:
This hover colour spec actually does nothing, because the headings don't have an associated `Action` with them, which means they have `Enabled = false`. The stuff from `OnClick()` should be moved to `Action`.

Additionally, the colour should also remain `Light1` after click, as per web.

![2021-06-04-142502_346x47_scrot](https://user-images.githubusercontent.com/20418176/120800831-b1699e80-c540-11eb-8bf5-089c0e87e461.png)

Reply:
Ah, I missed that.

Reply:
For now I can change the colour to `Light1` after click inside the `OnClick`. But I dont know how to change the colour back to `Light2` when the other entry is clicked.

Reply:
You'd probably have to somehow track the last item clicked at the level of the table of contents sidebar. Maybe not especially worth pursuing right now.

Reply:
Now for simplicity, I just remove the onclick and the colour change.